symengine/sympy_compat.py

```diff
@@ -43,6 +47,35 @@ def __new__(cls, i):
         return symengine.Integer(i)
 
 
+class Float(Number):
+    _classes = (symengine.RealDouble, symengine.RealMPFR)
+
+    def __new__(cls, num, dps=None, prec=None, precision=None):
+        if dps is not None and precision is not None:
+            raise ValueError('Both decimal and binary precision supplied. '
+                             'Supply only one. ')
+        if dps is None and precision is None:
+            dps = 15
+            if isinstance(num, Float):
+                return num
```

Comment:
`SymPy` sets `dps = 15` for cases where the argument is not an instance of `Float`. If the `num` is an instance of the `Float` class and no `dps` or `precision` is provided, it simply returns that same object. Should I change this? `dps = 15` evaluates to `precision = 53`, so should I return a RealDouble each time?